### DISCUSSION TITLE
How can I reference colors in animations?

### DISCUSSION THREAD

**Discussion by ackvf** [unocss/unocss#4273] [Q&A] [answered]
Hello, I would like to create new animation and specify a color from the theme, how can I do it?

```
const config = {
  theme: {
    colors: {
      ocean: {
        50: "#eefcfd",
        100: "#d3f4fa",
        200: "#ace8f5",
        300: "#73d6ed",
        400: "#33bbdd",
        500: "#1AB1DB",
        600: "#167ea4",
        700: "#196685",
        800: "#1d546d",
        900: "#1c465d",
        950: "#0d2d3f",
      },
    },
    animation: {
      keyframes: {
        actionButtonPulse: `
          {
            0%,
            100% {
              background-color: var(--un-color-ocean-700, inherit); // how to specify the color here?
            }
            50% {
              background-color: var(--un-color-ocean-500, inherit);
            }
          }
        `,
      },
      counts: {
        actionButtonPulse: "infinite",
      },
    },
  },
}
```

**Comment by zyyv** [comment, marked as answer]
Hi, we provide contextual themes in prelights. You can generate css variables for the theme key you need in prelights.

Refer to [Playground](https://unocss.dev/play/#html=DwEwlgbgBA7gTgQwA5IKZwHwCgpVJKAZzAC9UBaARgAYoEA7MAWwQBcKEBjVsAe3oBCAV1at%2BABSEAbQqigB6bMHngI2IA&config=JYWwDg9gTgLgBAbzgEwKYDNgDtUGEJaYDmcAvnOlBCHAOQCuWEAxgM6u0BQoksicYKKlaoYAVRHp6AGzIUqNBkzasAtIOGjV9STK6dUAD17w06AIYzTGbHgLEAFAk5w4rABbQYzejFYAuRBdXOAB3KHMwMFQoQNpQ1VZmIVQsOHdE5NRUimY82gAaYNIi1w0RP0CAbWCyoQqJDBknWpCNdGlgIndKinNpEVKQ0gBKIYBdIaMYVOQAFXdUEFQHGEXlkaCQuGYCVghpVAA6aQgiVfXUEeKhtaXUQOdt3dOoAK3tuBZUcyxH1u2AFYAAxxADE2XQzHQyEKAJCAEZgaC6GDkABmdAAFgscM%2BIQATMjweZmKgABzoQF4-FwdHE1EAdnRyAAbKhYUN8ViGbQwej0QAjQXITnw1wglF8hEAQQAQgiACJymn41m8sEI1mMn5Y1WfRkahEATlZrPJ1K5n3JRuQgKxrLFtLgxqNzAdgKdtONIPBwOQBIx6H1rhKrTD21%2BoHMMGABH%2B%2BIA1qgAJ6UczLd5PWmk2MEOW%2BGAEAAKMhEgQABuKQtnna5gQBSK3OpGNj51kKC0mJohURjIVQvaCBABu5igDlU2iwg4O0FU31%2BqkNwIKcGwiygwBgIwA3HAAPQH9IQUJwItuaLMYDoFPnxY7OdQdIxVAAfmr21In5CIIb7Y7OAu2YHs%2BywAch1iOAxwnKdGFnV4FzJJdJTXDcYm3Pcf1DH9v2dCtmzIQjdkYXpa0%2BXM4ywAsYCLLBSwGB46GwTAsG3VAQyI8MhgjAQhA6LoeneGptnI1xpHMFMYjiO5lk4ohRFwABlJTAice97jITYAF4AD4AO2IQYHoKA0grfwqAgeAxM%2BOCZ0gpCfhnFdAgAEgQWTjkg1gqloRcsFocZfJXQLSF3H87IQ%2Bd-NUSU3I8y4jm83z-MC3zJVC8LnVIKt8V40MJiKUZOCAA&css=PQKgBA6gTglgLgUzAYwK4Gc4HsC2YDCAyoWABYJQIA0YAhgHYAmYcUD6AZllDhWOqgAOg7nAB0YAGLcwCAB60cggDYIAXGBDAAUKDBi0mXGADe2sGC704AWgDuCGAHNScDQFYADJ4Dc5sAACtMLKAJ5gggCMLPK2ABR2pPBIcsoAlH4WAEa0yADWTlBYqEw2yFjK3Bpw5LxxAOTllVDoYpSMYgAs3vUZ2gC%2BmjpAA&options=N4IgLgTghgdgzgMwPYQLYgFwKgGzgUwBpxp5k0BhAVzjCVQoGVHNs8iTZEVUAJMVDla4CAXyA&version=0.64.0)

**Reply by ackvf** [reply]
Wow! Thank you!
So basically, the `theme.colors.ocean` object will not automatically generate `--un-color-ocean-{x}` values available in the stylesheet, I need to specify the needed variables myself, but I can still reference the theme in `getCSS()`, which is awesome. Is there something stopping me from mapping through the object?

```
Object.entries(theme['colors'].ocean).map(([k,v]) => ` --un-color-ocean-${k}: ${v} `).join('; ')
```
